src/ring_theory/fractional_ideal.lean

```diff
@@ -103,35 +101,41 @@ variables [algebra R P] [loc : is_localization S P]
 
 instance : has_coe (fractional_ideal S P) (submodule R P) := ⟨λ I, I.val⟩
 
-@[simp] lemma val_eq_coe (I : fractional_ideal S P) : I.val = I := rfl
+section set_like
 
-@[simp, norm_cast] lemma coe_mk (I : submodule R P) (hI : is_fractional S I) :
-  (subtype.mk I hI : submodule R P) = I := rfl
+instance : set_like (fractional_ideal S P) P :=
+{ coe := λ I, (I : submodule S P),
+  coe_injective' := set_like.coe_injective.comp subtype.coe_injective }
 
-instance : has_mem P (fractional_ideal S P) := ⟨λ x I, x ∈ (I : submodule R P)⟩
+lemma mem_val {I : fractional_ideal S P} {x : P} : x ∈ I.val ↔ x ∈ I := iff.rfl
```

Comment:
There's no real point in this lemma IMO, since `val_eq_coe` eliminates `val` anywhere it ends up appearing.

Reply:
I kept it just in case a `val` does appear. I should probably make a projection like `subtype.prop` since `I.2` still talks about `val`, then we don't need this shortcut lemma.